Improve responsive design on mobile devices

lin-lu opened the pull request.

chrisvfritz commented:
@lin-lu Can you please describe (and possibly print screenshots of) the problem you're seeing that you're trying to fix?

lin-lu commented:
The first problem is that when the screen width is between 720px and 900px, the hamburger menu button on the home page does not work. You can reproduce this bug by resizing the browser.

The second problem is that the search text box is displayed on top of the vue logo, when the width is around 800px. 
![screen shot 2016-10-01 at 17 37 25](https://cloud.githubusercontent.com/assets/11504510/19015134/811770a6-87fe-11e6-85d1-22e459cb1c41.png)
I changed the breakpoint of the media queries from 720px to 900px, so the mobile version will be displayed when the width is smaller than 900px.

After I changed that, I noticed that the max-width of the content is fixed at 600px, so when the screen width is larger than that and smaller than 900px, the content will appear on the left side of the page, leaving some empty space on the right side.
![screen shot 2016-10-01 at 17 57 12](https://cloud.githubusercontent.com/assets/11504510/19015266/3540350c-8801-11e6-941b-b37a93a8f55a.png)
To make it symmetry, I also centered the content for mobile devices.

chrisvfritz commented:
Thanks! This looks good. 👍
